### PR TITLE
[vLLM plugin] Refactor pooling model tests

### PR DESCRIPTION
**Pull request by mmanzoorTT**
### What's changed
- Move selected tests from push to nightly
- Reduce unnecessary compilation during warmup with modified configs
- Reuse model across multiple configs to leverage caching
- Remove redundant Qwen3-Embedding-4B 8K/16K tests (covered by perf)
- Create `run_pooling_test` to execute the models and removed duplicated code.
- General code cleanup

### CODE REVIEW

**Comment by AleksKnezevic** [comment]
I think we can probably limit to one model per architecture on push, so one embedding, one generative, eventually one Moe.

File: tests/integrations/vllm_plugin/pooling/test_qwen3_embedding.py [hunk 1]
```diff
@@ -11,6 +11,87 @@
...
+@pytest.mark.parametrize(
+    ["model_name", "baseline_path"],
+    [
+        pytest.param(
+            "Qwen/Qwen3-Embedding-0.6B",
+            "baseline/qwen3_embedding_0.6B_baseline.pt",
+        ),
+    ],
+)
+def test_embed_qwen3_push(
```
Comment (kmabeeTT) [inline tests/integrations/vllm_plugin/pooling/test_qwen3_embedding.py]: In general do not love the duplicated tests here that are 99% the same.  If we pulled the body to common/shared function `_run_embed_qwen3_test` and used by each test, then no duplication..

```

def _run_embed_qwen3_test(
    model_name: str,
    baseline_path: str,
    min_context_len: int = 32,
    experimental_enable_weight_bfp8_conversion: bool = False,
):
    """
    Test the Qwen3-Embedding models embedding outputs for correctness
    under different batching and padding scenarios.

    Test Setup:
    - Input consists of four prompts with token lengths [32, 15, 29, 7].
    - vLLM is configured with max_num_seqs=2, meaning each batch can contain
      up to 2 sequences and vLLM always process single prompt in first batch.
    - This results in three batches:
        1. First batch: first prompt (32 tokens) → no padding required.
        2. Second batch: second and third prompts concatenated (15 + 29 = 44
           tokens), padded to max_model_len=64.
        3. Third batch: fourth prompt (7 tokens), padded to max_model_len=32.

    Purpose:
    - Validates that the model produces embeddings consistent with precomputed
      baseline embeddings for each prompt.
    - Ensures Pearson Correlation Coefficient (PCC) > 0.99 for each embedding.

    Baseline Embeddings:
    - Baseline embeddings are computed using vLLM on CPU backend and stored in
      'baseline' directory.
    """
    path = os.path.join(os.path.dirname(__file__), baseline_path)
    loaded_data = torch.load(path)

    prompts = [
        "The quick-thinking engineer designed a compact neural processor that could adapt to changing data patterns in real time, optimizing energy use while maintaining exceptional computational accuracy as well.",
        "Hello, my name is chatbot. How can I help you?",
        "We build computers for AI. We design Graph Processors, high-performance RISC CPUs, and configurable chips that run our robust software stack.",
        "The capital of France is Paris",
    ]

    additional_config = {"min_context_len": min_context_len}
    if experimental_enable_weight_bfp8_conversion:
        additional_config["experimental_enable_weight_bfp8_conversion"] = True

    llm_args = {
        "model": model_name,
        "task": "embed",
        "dtype": "bfloat16",
        "max_model_len": 64,
        "disable_sliding_window": True,
        "max_num_batched_tokens": 64,
        "max_num_seqs": 2,
        "additional_config": additional_config,
    }
    model = vllm.LLM(**llm_args)

    output_embedding = model.embed(prompts)

    for idx, (prompt, output) in enumerate(zip(prompts, output_embedding)):
        embeds = output.outputs.embedding
        embeds_trimmed = (
            (str(embeds[:32])[:-1] + ", ...]") if len(embeds) > 32 else embeds
        )
        print(f"Prompt: {prompt!r} \nEmbeddings: {embeds_trimmed} (size={len(embeds)})")

        output_tensor = torch.tensor(embeds, dtype=torch.float32)
        golden_tensor = loaded_data[f"prompt{idx}"]
        pcc = torch.corrcoef(torch.stack([output_tensor, golden_tensor]))[0, 1]
        print("PCC:", pcc.item())
        assert pcc.item() > 0.99, f"PCC Error: Incorrect embedding for prompt{idx}"

        print("-" * 60)


@pytest.mark.push
@pytest.mark.single_device
@pytest.mark.parametrize(
    ["model_name", "baseline_path"],
    [
        pytest.param(
            "Qwen/Qwen3-Embedding-0.6B",
            "baseline/qwen3_embedding_0.6B_baseline.pt",
        ),
    ],
)
def test_embed_qwen3_push(model_name: str, baseline_path: str):
    _run_embed_qwen3_test(model_name, baseline_path)


@pytest.mark.nightly
@pytest.mark.single_device
@pytest.mark.parametrize(
    ["model_name", "baseline_path", "experimental_enable_weight_bfp8_conversion"],
    [
        pytest.param(
            "Qwen/Qwen3-Embedding-4B",
            "baseline/qwen3_embedding_4B_baseline.pt",
            False,
        ),
        pytest.param(
            "Qwen/Qwen3-Embedding-8B",
            "baseline/qwen3_embedding_8B_baseline.pt",
            True,
            marks=pytest.mark.xfail(
                reason="Static CBs exceed L1 size - https://github.com/tenstorrent/tt-xla/issues/2935"
            ),
        ),
    ],
)
@pytest.mark.parametrize("min_context_len", [32, 64])
def test_embed_qwen3_nightly(
    model_name: str,
    baseline_path: str,
    experimental_enable_weight_bfp8_conversion: bool,
    min_context_len: int,
):
    _run_embed_qwen3_test(
        model_name,
        baseline_path,
        min_context_len=min_context_len,
        experimental_enable_weight_bfp8_conversion=experimental_enable_weight_bfp8_conversion,
    )


```

**Comment by AleksKnezevic** [comment]
Looks good. Thanks @mmanzoorTT
